com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/TimeProfiler.java

```diff
@@ -225,14 +197,25 @@ public static void createReport(Boolean fromEditor) {
                         System.out.println("Failed to create directories: " + parentDir);
                     }
                 }
-                if (extension.equals(".json")) {
-                    saveJSON(jsonReport, finalReportFile);
-                }
-                else if (extension.equals(".html")) {
-                    saveHTML(jsonReport, finalReportFile);
+                int bufferSize = 8 * 1024 * 1024; // 8 MB in bytes
+                try (BufferedWriter writer = new BufferedWriter(new FileWriter(finalReportFile), bufferSize)) {
+                    if (extension.equals(".json")) {
+                        generateJSON(_rootScope, writer);
+                    } else if (extension.equals(".html")) {
+                        String[] htmlContent = getHtmlContent();
+                        writer.write(htmlContent[0]);
+                        writer.flush();
+                        generateJSON(_rootScope, writer);
+                        writer.flush();
+                        writer.write(htmlContent[1]);
```

Comment:
Add the first "before" html part of the time report, then write json, then add the second part of the html template.